group-coordinator/src/main/java/org/apache/kafka/coordinator/group/taskassignor/TaskId.java

```diff
@@ -63,4 +63,9 @@ public String toString() {
             ", partition=" + partition +
             '}';
     }
+
+    public int compareTo (TaskId other) {
```

Comment:
This is not a correct way to implement the comparison, I think.

Please check https://stackoverflow.com/questions/369512/how-to-compare-objects-by-multiple-fields or something like that.